code/datums/components/nanites.dm

```diff
@@ -0,0 +1,163 @@
+/datum/component/nanites
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	var/mob/living/host_mob
+	var/nanite_volume = 100		//amount of nanites in the system, used as fuel for nanite programs
+	var/max_nanites = 500		//maximum amount of nanites in the system
+	var/regen_rate = 0.5		//nanites generated per second
+	var/safety_threshold = 50	//how low nanites will get before they stop processing/triggering
+	var/cloud_id = 0 			//0 if not connected to the cloud, 1-100 to set a determined cloud backup to draw from
+	var/next_sync = 0
+	var/list/datum/nanite_program/programs = list()
+	var/max_programs = 15
+
+	var/stealth = FALSE //if TRUE, does not appear on HUDs and health scans, and does not display the program list on nanite scans
+
+/datum/component/nanites/Initialize(amount)
+	nanite_volume = amount
+
+	//Nanites without hosts are non-interactive through normal means
+	if(isliving(parent))
```

Comment:
Ignore this exact code actually since I think I noticed you had the component on a machine occasionally. You still want to return COMPONENT_INCOMPATIBLE though if the component is on an unsupported type.

Reply:
can do